### ISSUE TITLE
Integrating an arion-compose into a NixOS config?

### ISSUE THREAD

**Issue by Nebucatnetzer** [hercules-ci/arion#146]
Is it possible to define a docker environment in a NixOS config so that it will get deployed when you rebuild the system?

**Comment by roberth** [comment]
Yes, the flake comes with `nixosModules.arion` for this purpose. That's [this file](https://github.com/hercules-ci/arion/blob/master/nixos-module.nix).

**Comment by Nebucatnetzer** [comment]
Ah okay so it would then be something like this?

```
virtualisation.arion = {
  enable = true;
  backend = “docker”;
  settings = {
      services.postgres = {
        service.image = "postgres:10";
        service.volumes = [ "${toString ./.}/postgres-data:/var/lib/postgresql/data" ];
        service.environment.POSTGRES_PASSWORD = "mydefaultpass";
      };
  };
```

**Comment by roberth** [comment]
That looks alright, but note that the password will be readable in the nix store.

**Comment by roberth** [comment]
While you can still technically use `docker-compose` with its own environment variable substitution, I wouldn't recommend it because that functionality may go away.
I prefer to use module system abstractions for this: user-defined top level `options` and `imports = [ ./environment.nix ];`, that kind of thing.

**Comment by Nebucatnetzer** [comment]
Something like the `passwordFile` option from restic might be an idea?
```
service.environmentFile = “/etc/docker-compose/nextcloud_env”
```
